velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -1,802 +0,0 @@
-    carryToLeft = true;
-  } else {
-    fraction = aFractionScaled + bFractionScaled;
-  }
-
-  // Scale up the whole part and scale down the fraction part to combine them.
-  fraction = reduceScale(TResult(fraction), higherScale - rScale);
-  const auto whole = TResult(aWhole) + TResult(bWhole) + TResult(carryToLeft);
-  return decimalAddResult(whole, TResult(fraction), rScale, overflow);
-}
-
-// Adds two opposite values by adding the whole and fraction parts separately.
-template <typename TResult, typename A, typename B>
-inline static TResult addLargeOpposite(
-    A a,
-    B b,
-    uint8_t aScale,
-    uint8_t bScale,
-    int32_t rScale,
-    bool& overflow) {
-  VELOX_DCHECK(
-      (a < 0 && b > 0) || (a > 0 && b < 0),
-      "One positve and one negative value are expected in addLargeOpposite.");
-
-  // Separate whole and fraction parts.
-  const auto [aWhole, aFraction] = getWholeAndFraction<A>(a, aScale);
-  const auto [bWhole, bFraction] = getWholeAndFraction<B>(b, bScale);
-
-  // Adjust fractional parts to higher scale.
-  const auto higherScale = std::max(aScale, bScale);
-  const auto aFractionScaled =
-      increaseScale((int128_t)aFraction, higherScale - aScale);
-  const auto bFractionScaled =
-      increaseScale((int128_t)bFraction, higherScale - bScale);
-
-  // No need to consider overflow because two inputs are opposite.
-  int128_t whole = (int128_t)aWhole + (int128_t)bWhole;
-  int128_t fraction = aFractionScaled + bFractionScaled;
-
-  // If the whole and fractional parts have different signs, adjust them to the
-  // same sign.
-  const auto scaleFactor = velox::DecimalUtil::kPowersOfTen[higherScale];
-  if (whole < 0 && fraction > 0) {
-    whole += 1;
-    fraction -= scaleFactor;
-  } else if (whole > 0 && fraction < 0) {
-    whole -= 1;
-    fraction += scaleFactor;
-  }
-
-  // Scale up the whole part and scale down the fraction part to combine them.
-  fraction = reduceScale(TResult(fraction), higherScale - rScale);
-  return decimalAddResult(TResult(whole), TResult(fraction), rScale, overflow);
-}
-
-template <typename TResult, typename A, typename B>
-inline static TResult addLarge(
-    A a,
-    B b,
-    uint8_t aScale,
-    uint8_t bScale,
-    int32_t rScale,
-    bool& overflow) {
-  if (a >= 0 && b >= 0) {
-    // Both non-negative.
-    return addLargeNonNegative<TResult, A, B>(
-        a, b, aScale, bScale, rScale, overflow);
-  } else if (a <= 0 && b <= 0) {
-    // Both non-positive.
-    return TResult(-addLargeNonNegative<TResult, A, B>(
-        A(-a), B(-b), aScale, bScale, rScale, overflow));
-  } else {
-    // One positive and the other negative.
-    return addLargeOpposite<TResult, A, B>(
-        a, b, aScale, bScale, rScale, overflow);
-  }
-}
-
-template <
-    typename R /* Result Type */,
-    typename A /* Argument1 */,
-    typename B /* Argument2 */,
-    typename Operation /* Arithmetic operation */>
-class DecimalBaseFunction : public exec::VectorFunction {
```

Comment:
This PR also changes vector functions as simple functions. Does it make sense to do the vector-to-simple refactor in a separate PR, and support allowPrecisionLoss in this one?

Reply:
Do you agree? If you think it is OK, I will only use `SparkRegistrationConfig` in this PR and revert to vector function. @mbasmanova

Reply:
Perhaps we don't need to change back to vector function in this PR. After the separate PR lands, we can do a rebase to it.